### ISSUE TITLE
GPU Temperature Monitoring

### ISSUE THREAD

**Issue by SETIguy** [BOINC/boinc#1379]
I'd like to see GPU temperature monitoring and temperature limits in the desktop client similar to the battery temperature limits seen in the Android client. This may require an additional package or library to be installed.  Response to overtemp could be a pause in processing, or underclocking the GPU.

**Comment by runninghyk** [comment]
Maybe Windows can use this
`https://github.com/openhardwaremonitor/openhardwaremonitor`

**Comment by cminnoy** [comment]
NVIDIA Linux:
                 const nvmlTemperatureSensors_t sensorType = NVML_TEMPERATURE_GPU;
                unsigned int temperature;
                nvmlDeviceGetTemperature(m_handle, sensorType, &temperature);

**Comment by AenBleidd** [comment]
@cminnoy, but it's only for nvidia and linux. But BOINC also could run on Windows, OSX, and run tasks on AMD and Intel GPUs.
So we should have here a solution for all (or almost all) configurations that user can have

**Comment by cminnoy** [comment]
plug-ins...life is all about plug-ins

**Comment by AenBleidd** [comment]
I don't know. Plugins were popular 20 years ago. Nowadays plugins are used in software that should be customized (e.g. Wordpress) or when you need to enable what you need only and disable everything else.

**Comment by cminnoy** [comment]
No, plug-ins allow your software to evolve more quickly. It is a form of separation of concern.

https://en.wikipedia.org/wiki/Plug-in_(computing)
https://en.wikipedia.org/wiki/Separation_of_concerns

**Comment by CharlieFenton** [comment]
Years ago, I added code in _client/hostinfo_unix.cpp_ `int get_max_cpu_temperature()`to get CPU temperature on PowerPC and Intel Macs. It was never called because no one ever added the corresponding code to get the temperature on other platforms, so no use of temperature information was ever incorporated in BOINC. Apparently, someone removed it at some point because it was not being used. 

I have no idea whether that code still works on modern Macs.

**Comment by yuleeque** [comment]
> Years ago, I added code in client/hostinfo_unix.cpp int get_max_cpu_temperature() ...

Once Ernest Hemingway bet that he would write the shortest story that could touch anyone.
